HW_4/task.py

```diff
@@ -0,0 +1,115 @@
+'''
+Написать программу, которая скачивает изображения с заданных URL-адресов и сохраняет их на диск. Каждое изображение должно сохраняться в отдельном файле, название которого соответствует названию изображения в URL-адресе.
+Например, URL-адрес: https://example/images/image1.jpg -> файл на диске: image1.jpg
+— Программа должна использовать многопоточный, многопроцессорный и асинхронный подходы.
+— Программа должна иметь возможность задавать список URL-адресов через аргументы командной строки.
+— Программа должна выводить в консоль информацию о времени скачивания каждого изображения и общем времени выполнения программы.
+'''
+
+import os
+from pathlib import Path
+import requests
+import time
+import argparse
+import threading
+from multiprocessing import Process
+import asyncio
+import aiohttp
+
+images = []
+with open(r'HW_4/images.txt') as f:
+    for image in f.readlines():
```

Comment:
Файловый дескриптор уже является итератором.
```suggestion
    for image in f:
```